src/script.ts

```diff
@@ -0,0 +1,135 @@
+/*
+  Copyright 2019 Esri
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import { loadCss } from './utils/css';
+import utils from './utils/index';
+import { getCdnUrl } from './utils/url';
+
+function createScript(url) {
+  const script = document.createElement('script');
+  script.type = 'text/javascript';
+  script.src = url;
+  script.setAttribute('data-esri-loader', 'loading');
+  return script;
+}
+
+// add a one-time load handler to script
+// and optionally add a one time error handler as well
+function handleScriptLoad(script, callback, errback?) {
+  let onScriptError;
+  if (errback) {
+    // set up an error handler as well
+    onScriptError = handleScriptError(script, errback);
+  }
+  const onScriptLoad = () => {
+    // pass the script to the callback
+    callback(script);
+    // remove this event listener
+    script.removeEventListener('load', onScriptLoad, false);
+    if (onScriptError) {
+      // remove the error listener as well
+      script.removeEventListener('error', onScriptError, false);
+    }
+  };
+  script.addEventListener('load', onScriptLoad, false);
+}
+
+// add a one-time error handler to the script
+function handleScriptError(script, callback) {
+  const onScriptError = (e) => {
+    // reject the promise and remove this event listener
+    callback(e.error || new Error(`There was an error attempting to load ${script.src}`));
+    // remove this event listener
+    script.removeEventListener('error', onScriptError, false);
+  };
+  script.addEventListener('error', onScriptError, false);
+  return onScriptError;
+}
+
+// interfaces
+export interface ILoadScriptOptions {
+  version?: string;
+  url?: string;
+  css?: string | boolean;
+  dojoConfig?: { [propName: string]: any };
+  insertCssBefore?: string;
+}
+
+// get the script injected by this library
+export function getScript() {
+  return document.querySelector('script[data-esri-loader]') as HTMLScriptElement;
+}
+// has ArcGIS API been loaded on the page yet?
```

Comment:
good opportunity for one liner here.

```ts
export const getScript = () = document.querySelector('script[data-esri-loader]') as HTMLScriptElement;
```

Reply:
good idea, but I'm tired of waiting for Travis